src/NodaTime.Test/DateIntervalTest.cs

```diff
@@ -208,5 +208,274 @@ public void Deconstruction()
                 Assert.AreEqual(end, actualEnd);
             });
         }
+
+        [Test]
+        public void Contains_NullInterval_Throws()
+        {
+            var start = new LocalDate(2017, 11, 6);
+            var end = new LocalDate(2017, 11, 10);
+            var value = new DateInterval(start, end);
+
+            Assert.That(() => value.Contains(null), Throws.TypeOf<ArgumentNullException>());
+        }
+
+        [Test]
+        public void Contains_IntervalWithinAnotherCalendar_Throws()
+        {
+            var value = new DateInterval(
+                new LocalDate(2017, 11, 6, CalendarSystem.Gregorian),
+                new LocalDate(2017, 11, 10, CalendarSystem.Gregorian));
+
+            var other = new DateInterval(
+                new LocalDate(2017, 11, 6, CalendarSystem.Coptic),
+                new LocalDate(2017, 11, 10, CalendarSystem.Coptic));
+
+            Assert.That(() => value.Contains(other), Throws.TypeOf<ArgumentException>());
+        }
+
+        [Test]
+        public void Contains_EqualInterval()
+        {
+            var start = new LocalDate(2014, 3, 7);
+            var end = new LocalDate(2014, 3, 7);
+            var value = new DateInterval(start, end);
+            var same = new DateInterval(start, end);
+
+            Assert.True(value.Contains(same));
+        }
+
+        [Test]
+        public void Contains_LaterInterval()
+        {
+            var value = new DateInterval(new LocalDate(2014, 3, 7), new LocalDate(2014, 3, 31));
+            var other = new DateInterval(new LocalDate(2015, 1, 1), new LocalDate(2015, 4, 1));
+
+            Assert.False(value.Contains(other));
+        }
+
+        [Test]
+        public void Contains_EarlierInterval()
+        {
+            var value = new DateInterval(new LocalDate(2014, 3, 7), new LocalDate(2014, 3, 31));
+            var other = new DateInterval(new LocalDate(2013, 6, 26), new LocalDate(2013, 7, 25));
+
+            Assert.False(value.Contains(other));
+        }
+
+        [Test]
```

Comment:
We could potentially use parameterized tests for all of these. I'd probably take strings representing the start/end intervals, and the expected results. (Strings rather than lots of integers to help with readability.)

Reply:
Got it.